src/util/codec/mysql/time.rs

```diff
@@ -230,18 +235,28 @@ impl Time {
 
         let frac = try!(parse_frac(frac_str.as_bytes(), fsp));
         if y == 0 && m == 0 && d == 0 && h == 0 && minute == 0 && sec == 0 {
-            return Ok(zero_datetime());
+            return Ok(zero_datetime(tz));
         }
         if y < 0 || y > 9999 {
             return Err(box_err!("unsupport year: {}", y));
         }
-        let t = try!(ymd_hms_nanos(y, m, d, h, minute, sec, frac * TEN_POW[9 - fsp as usize]));
+        let t = try!(ymd_hms_nanos(tz,
```

Comment:
No, it's not a `NaiveDateTime`. And I don't think add any type related information to name is a good idea.